diff --git a/subprojects/gst-plugins-base/meson.build b/subprojects/gst-plugins-base/meson.build
index 1111111..2222222 100644
--- a/subprojects/gst-plugins-base/meson.build
+++ b/subprojects/gst-plugins-base/meson.build
@@ -578,7 +578,9 @@ pkgconfig.generate(
 subdir('gst-libs')
 subdir('gst')
 subdir('ext')
-subdir('docs')
+if get_option('doc').allowed()
+  subdir('docs')
+endif
 subdir('tests')
 subdir('pkgconfig')

diff --git a/subprojects/gst-plugins-bad/meson.build b/subprojects/gst-plugins-bad/meson.build
index 1111111..2222222 100644
--- a/subprojects/gst-plugins-bad/meson.build
+++ b/subprojects/gst-plugins-bad/meson.build
@@ -644,7 +644,9 @@ subdir('gst')
 subdir('sys')
 subdir('ext')
 subdir('tests')
-subdir('docs')
+if get_option('doc').allowed()
+  subdir('docs')
+endif
 subdir('pkgconfig')

 # This is hardcoded as disabled. If someone wants to build this they
